cs_tools/sync/postgres/syncer.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import annotations
+
+import pydantic
+import sqlalchemy as sa
+import sqlmodel
+from cs_tools.sync.base import DatabaseSyncer
+from typing import TYPE_CHECKING, Any, Literal ,Optional
+from pydantic_core import PydanticCustomError
+import pathlib
+import logging
+from sqlalchemy.orm import sessionmaker
+from cs_tools import __version__
+from cs_tools.sync import utils as sync_utils
+from cs_tools.sync.base import DatabaseSyncer
+
+# if TYPE_CHECKING:
+from cs_tools.sync.types import TableRows
+
+log = logging.getLogger(__name__)
+
+
+class Postgres(DatabaseSyncer):
+    """
+    Interact with Postgres Database
+    """
+    
+    __manifest_path__ = pathlib.Path(__file__).parent / "MANIFEST.json"
+    __syncer_name__ = "postgres"
+   
+    dbname: str 
+    user:  str 
+    password:  str 
+    host:  str 
+    port: int 
+    schema_ : str = pydantic.Field(default="PUBLIC",alias="schema")
+    authentication: Literal["basic", "key-pair", "sso", "oauth"]
```

Comment:
Did you intend to offer multiple authentication types? I've only ever configured basic username/password for pg, so I am not sure if any other auth methods are standard. If you want to leave a spot for this to be open to expansion, we can copy **[the Redshift implementation](https://github.com/thoughtspot/cs_tools/blob/master/cs_tools/sync/redshift/syncer.py#L29)**.

Reply:
Postgres does not offer itself with "key:pair", "sso" or "oauth" authentication but can be implemented with third party like Google, Github. Also, Postgres offers LDAP Server or GSSAPI authentication which if needed can be developed later. Hence, I removed this line for now and added comment to develop authentication later based on the scope.